neurolib/models/model.py

```diff
@@ -179,7 +177,23 @@ def run(
                 logging.warn(f"{self.name}: BOLD model not initialized, not simulating BOLD. Use `run(bold=True)`")
                 bold = False
             self.integrateChunkwise(chunksize=chunksize, bold=bold, append_outputs=append)
-            return
+
+        # check if there was a problem with the simulated data
+        self.checkOutputs()
+
+    def checkOutputs(self):
+        # check nans in output
+        if np.isnan(np.sum(self.output)):
```

Comment:
```suggestion
        if np.isnan(self.output).any():
```
maybe cleaner? not sure...

Reply:
Yes, thanks for making me aware of this. This is the way!